peer.go

```diff
@@ -781,15 +791,43 @@ out:
 			p.htlcManMtx.Unlock()
 
 			p.wg.Add(1)
-			go p.htlcManager(newChanReq.channel, plexChan, downstreamLink, upstreamLink)
+			go p.htlcManager(newChanReq.channel, plexChan,
+				downstreamLink, upstreamLink)
 
 			close(newChanReq.done)
 
 		case req := <-p.localCloseChanReqs:
+			localCloseInitsMtx.Lock()
+			localCloseInits[*req.chanPoint] = req
+			localCloseInitsMtx.Unlock()
+
 			p.handleLocalClose(req)
 
-		case req := <-p.remoteCloseChanReqs:
-			p.handleRemoteClose(req)
+		case req := <-p.shutdownChanReqs:
+			localCloseInitsMtx.RLock()
+			_, ok := localCloseInits[req.ChannelID]
+			localCloseInitsMtx.RUnlock()
+			if !ok {
+				// we're the responder.
+				p.handleShutdownResponse(req)
+			}
+
+		case req := <-p.closingSignedChanReqs:
+			localCloseInitsMtx.RLock()
+			localCloseReq, ok := localCloseInits[req.ChannelID]
+			localCloseInitsMtx.RUnlock()
+			if ok {
+				// we're the initiator
+				p.handleInitClosingSigned(localCloseReq, req)
+
+				// this won't work if there's fee disagreement
+				localCloseInitsMtx.Lock()
+				delete(localCloseInits, req.ChannelID)
```

Comment:
From my reading, the pending close entry needs to be deleted from the map unconditionally, otherwise we won't be able to reclaim memory and also won't be able to properly handle retransmits.